### PR TITLE
Uses a fixed Rust version in CI

### PR DESCRIPTION
**Pull request by eadventurous**
Signed-off-by: Egor Ivkov <e.o.ivkov@gmail.com>

### Description of the Change
Uses a fixed Rust version in CI. This PR fixes the last places where not a fixed, but latest rust version was used.

### Benefits
We will migrate to new version explicitly from now on. This will prevent version migration failures in unrelated to the migration PRs.

### CODE REVIEW

File: .github/workflows/iroha2-dev-pr.yml [hunk 1]
```diff
@@ -128,7 +128,7 @@ jobs:
       - name: Install latest rust
         uses: actions-rs/toolchain@v1
         with:
-          toolchain: stable
+          toolchain: 1.52
```
Comment (Revertron) [inline .github/workflows/iroha2-dev-pr.yml:131]: Can we use 1.52.1?
As 1.52 sometimes produced faulty binaries :(
https://blog.rust-lang.org/2021/05/10/Rust-1.52.1.html

**Review by Revertron — CHANGES_REQUESTED** [review]

File: .github/workflows/iroha2-dev-pr.yml (hunk 1 above)
Comment (eadventurous) [reply .github/workflows/iroha2-dev-pr.yml:131]: This will install the latest 1.52.x

Comment (eadventurous) [reply .github/workflows/iroha2-dev-pr.yml:131]: Which means that all the hotfixes will be available instantly, but the breaking changes will need to be migrated manually.

Comment (Revertron) [reply .github/workflows/iroha2-dev-pr.yml:131]: Ah, okay then!

**Review by Revertron — APPROVED** [review]

**Review by i1i1 — APPROVED** [review]

**Pull request merged**